pkg/storage/bucket/azure/bucket_client.go

```diff
@@ -21,7 +21,6 @@ func NewBucketClient(cfg Config, name string, logger log.Logger) (objstore.Bucke
 	bucketConfig.ContainerName = cfg.ContainerName
 	bucketConfig.Endpoint = cfg.Endpoint
 	bucketConfig.MaxRetries = cfg.MaxRetries
-	bucketConfig.MSIResource = cfg.MSIResource
```

Comment:
Was this removed? If so, then I guess we have to deprecate `MSIResource` from Mimir config, because not used anymore.

Reply:
The struct field was not removed, but was marked as deprecated in `thanos-io/objstore` and doesn't do anything if you set it.
I changed the code here to use `DeprecatedFlag`.